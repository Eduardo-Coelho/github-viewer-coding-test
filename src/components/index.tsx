export * from './commitDetails/CommitDetails';
export * from './commitTable/CommitTable';
export * from './repoSearch/RepoSearch';
export * from './loader/Loader';
export * from './modal/Modal';
