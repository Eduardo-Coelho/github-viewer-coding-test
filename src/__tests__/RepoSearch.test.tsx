import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { RepoSearch } from '../components';

test('RepoSearch matches snapshot', () => {
  const { asFragment } = render(<RepoSearch onSubmit={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test('RepoSearch form submission calls onSubmit with owner and repo', () => {
  const handleSubmit = jest.fn();
  const { getByPlaceholderText, getByText } = render(
    <RepoSearch onSubmit={handleSubmit} />,
  );

  const ownerInput = getByPlaceholderText('GitHub Owner');
  const repoInput = getByPlaceholderText('GitHub Repo');
  const submitButton = getByText('Search');

  fireEvent.change(ownerInput, { target: { value: 'facebook' } });
  fireEvent.change(repoInput, { target: { value: 'react' } });
  fireEvent.click(submitButton);

  expect(handleSubmit).toHaveBeenCalledWith('facebook', 'react');
});

test('RepoSearch form submission is disabled if inputs are empty', () => {
  const handleSubmit = jest.fn();
  const { getByText } = render(<RepoSearch onSubmit={handleSubmit} />);

  const submitButton = getByText('Search');
  fireEvent.click(submitButton);

  expect(handleSubmit).not.toHaveBeenCalled();
});

test('RepoSearch form inputs update correctly', () => {
  const { getByPlaceholderText } = render(<RepoSearch onSubmit={() => {}} />);

  const ownerInput: any = getByPlaceholderText('GitHub Owner');
  const repoInput: any = getByPlaceholderText('GitHub Repo');

  fireEvent.change(ownerInput, { target: { value: 'microsoft' } });
  fireEvent.change(repoInput, { target: { value: 'vscode' } });

  expect(ownerInput.value).toBe('microsoft');
  expect(repoInput.value).toBe('vscode');
});
