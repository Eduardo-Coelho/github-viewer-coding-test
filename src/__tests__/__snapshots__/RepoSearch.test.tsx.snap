// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoSearch matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="repo-form"
  >
    <input
      placeholder="GitHub Owner"
      type="text"
      value=""
    />
    <input
      placeholder="GitHub Repo"
      type="text"
      value=""
    />
    <button
      disabled=""
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
